app/forms/training_details_form.rb

```diff
@@ -68,6 +69,10 @@ def commencement_date_valid
     end
   end
 
+  def commencement_date_year_is_four_digits
+    errors.add(:commencement_date, :invalid_year) if commencement_date.is_a?(Date) && commencement_date.year.digits.length != 4
```

Comment:
Why would `commencement_date` not be a date? is it just if it might be nil?

Reply:
oh I see, it gets turned into a struct when it is invalid, I presume so it can display on the page again with errors 👍